### DISCUSSION TITLE
Color and Distance Sensor question

### DISCUSSION THREAD

**Discussion by Artuchu** [pybricks/support#1974] [Q&A] [answered]
Hello, so I'm trying to copy a program I made over onto pybricks and a I'm wondering how to set up a Color and Distance sensor to be used both for sensing colors and running IR programs. On the Powered up app you can do that by swapping what mode the sensor is in with modes 7 and 8, however I keep running into Invalid argument errors whenever I try to use the setup for that. I've tried a few different methods to try to get this error to go away and nothing has seamed to work, and I've been looking for people with this same issue and I'm not finding anything.

The reason I'm trying to do it this way is because this is eventually going to become a GBC module, and we already have a bunch of 9-volt/power function's system things and so this would seriously simplify a lot of electronics, thank you guys for your time!

![image](https://github.com/user-attachments/assets/aa157b02-f6ff-443c-8a68-a722eef8f723)
![image](https://github.com/user-attachments/assets/cd39220b-69b2-4cb4-86ed-5bca16528ed3)
![image](https://github.com/user-attachments/assets/bbfa706d-7ee1-411d-9b24-2a72feb0fca7)

**Comment by BertLindeman** [comment]
Hello,

You can simply use the distance and the color from the sensor:

![image](https://github.com/user-attachments/assets/49e73b9b-7381-46e7-9b61-4e3db72c6646)

Output:
```
20
Color.NONE
```

Bert

**Reply by Artuchu** [reply]
I would but I need to swap between using it for color sensing and sending IR signals, and all honesty I don't know how distance sensing is going to help here

**Reply by BertLindeman** [reply]
Ah, oops, I do not know that either, sorry.
I even did not know it could....

**Reply by BertLindeman** [reply]
Indeed a nice way to control the PFmotor. 

Did not really  setup the pfmotor on the IR sensor, but the pybricks blocks allow this:
![image](https://github.com/user-attachments/assets/0d6cd4d8-5ae7-4e55-8921-4ac50f88ac6a)

No need to switch the sensor mode. 

Bert

**Comment by laurensvalk** [comment]
You've run into a bit of a subtle bug, thanks for reporting!

The power function motor is expecting Color.RED or Color.BLUE for the port choice, but both of those are overridden in your program and the motor class doesn't know what to do with your custom values.

As a workaround, try renaming your custom colors to `MyRed`, for example.

This is not related to mode switching. That all happens automatically.

**Reply by BertLindeman** [reply, marked as answer]
Rename of the custom colors (of course) helps.
Also on the other channels.

A simple program:
![image](https://github.com/user-attachments/assets/5cc9a726-39e3-4277-8a90-84cffc7499e7)

**Reply by laurensvalk** [reply]
Thank you for adding a working example @BertLindeman  - much appreciated.

**Comment by Artuchu** [comment]
That seamed to work! thank you! i'll let everyone know if any other issues arise
